xdsl/dialects/riscv.py

```diff
@@ -346,13 +351,55 @@ def parse(cls, parser: Parser) -> Self:
             regions=regions,
         )
 
+    @classmethod
+    def parse_args(cls, parser: Parser) -> Sequence[UnresolvedOperand]:
+        if arg := parser.parse_optional_unresolved_operand():
+            args = [arg]
+            while parser.parse_optional_punctuation(",") and (
+                arg := parser.parse_optional_unresolved_operand()
+            ):
+                args.append(arg)
+            return args
+        return []
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        return parser.parse_optional_attr_dict()
+
     def print(self, printer: Printer) -> None:
-        printer.print_operands(self.operands)
+        first = True
+        first &= not self.print_operands(printer, first)
```

Comment:
what does first mean? it feels like this logic could be shifted out of the helper methods, and be here for clarity, even if that means checking for an empty sequence twice